spec/unit/puppet/provider/rabbitmq_parameter/rabbitmqctl_federation_spec.rb

```diff
@@ -0,0 +1,74 @@
+require 'spec_helper'
+
+provider_class = Puppet::Type.type(:rabbitmq_parameter).provider(:rabbitmqctl)
+describe provider_class do
+  let(:resource) do
+    Puppet::Type.type(:rabbitmq_parameter).new(
+      name: 'documentumFederation@/',
+      component_name: 'federation',
+      value: {
+        'uri'      => 'amqp://',
+        'expires'  => '360000'
+      }
+    )
+  end
+  let(:provider) { provider_class.new(resource) }
+
+  after do
+    described_class.instance_variable_set(:@parameters, nil)
+  end
+
+  describe '#prefetch' do
+    it 'exists' do
+      expect(described_class).to respond_to :prefetch
+    end
+
+    it 'matches' do
+      provider_class.expects(:rabbitmqctl_list).with('vhosts').returns <<-EOT
```

Comment:
In our modules we've been moving away from setting `expects` before the test but rather use `allow()`, then the test and then `expect().to have_received()`.
https://relishapp.com/rspec/rspec-mocks/docs/basics/spies#spy-on-a-method-on-a-partial-double

Reply:
Makes sense; given that the submitter said they’re not very experienced with tests, and that this matches the style of the rest of the tests, I’m inclined to say let’s accept this, and then someone can refactor the tests to the newer style later if it makes sense?

Reply:
I'm fine with that.